core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -990,24 +990,20 @@ class KafkaApis(val requestChannel: RequestChannel,
         val bandwidthThrottleTimeMs = quotas.fetch.maybeRecordAndGetThrottleTimeMs(request, responseSize, timeMs)
 
         val maxThrottleTimeMs = math.max(bandwidthThrottleTimeMs, requestThrottleTimeMs)
+        unconvertedFetchResponse = fetchContext.updateAndGenerateResponseData(partitions)
+
         // [LIKAFKA-45345] even if the throttleTimeMs is 0, we still record it so that
         // the throttle-time sensor does not expire before the byte-rate sensor in quotas.fetch
         // or the request-time sensor in quotas.request.
```

Comment:
Please add a link to the current ticket (whatever it is) and KIP-215(?), and describe the recent issue and change in behavior. (Or add a new comment a couple lines up.)